examples/parachain-example/src/lib.rs

```diff
@@ -0,0 +1,368 @@
+            mortality_hash,
+        };
+        Self {
+            extra_params,
+            additional_params,
+        }
+    }
+
+    fn encode_extra_to(&self, v: &mut Vec<u8>) {
+        self.extra_params.encode_to(v);
+    }
+
+    fn encode_additional_to(&self, v: &mut Vec<u8>) {
+        self.additional_params.encode_to(v);
+    }
+}
+
+////////////////////////////////////////////////////////////
+// Second Config (just using subxt types)
+////////////////////////////////////////////////////////////
+
+pub enum StatemintConfig2 {}
+
+impl Config for StatemintConfig2 {
+    type Index = u32;
+    type Hash = subxt::utils::H256;
+    type AccountId = subxt::utils::AccountId32;
+    type Address = subxt::utils::MultiAddress<Self::AccountId, ()>;
+    type Signature = subxt::utils::MultiSignature;
+    type Hasher = subxt::config::substrate::BlakeTwo256;
+    type Header = subxt::config::substrate::SubstrateHeader<u32, Self::Hasher>;
+    type ExtrinsicParams = StatemintExtrinsicParams;
+}
+
+#[derive(Encode, Debug, Clone)]
+pub struct StatemintExtrinsicParams2 {
+    extra_params: StatemintExtraParams2,
+    additional_params: StatemintAdditionalParams2,
+}
+
+#[derive(Encode, Debug, Clone)]
+pub struct StatemintExtraParams2 {
+    era: subxt::config::extrinsic_params::Era,
+    nonce: u32,
+    charge: subxt::config::substrate::AssetTip,
+}
+
+#[derive(Encode, Debug, Clone)]
+pub struct StatemintAdditionalParams2 {
+    spec_version: u32,
+    tx_version: u32,
+    genesis_hash: subxt::utils::H256,
+    mortality_hash: subxt::utils::H256,
+}
+
+impl ExtrinsicParams<<StatemintConfig2 as Config>::Index, <StatemintConfig2 as Config>::Hash>
+    for StatemintExtrinsicParams2
+{
+    /// mortality hash, era, charge
+    type OtherParams = (
+        subxt::utils::H256,
+        subxt::config::extrinsic_params::Era,
+        subxt::config::substrate::AssetTip,
+    );
+
+    fn new(
+        spec_version: u32,
+        tx_version: u32,
+        nonce: <StatemintConfig2 as Config>::Index,
+        genesis_hash: <StatemintConfig2 as Config>::Hash,
+        other_params: Self::OtherParams,
+    ) -> Self {
+        let (mortality_hash, era, charge) = other_params;
+
+        let extra_params = StatemintExtraParams2 { era, nonce, charge };
+
+        let additional_params = StatemintAdditionalParams2 {
+            spec_version,
+            tx_version,
+            genesis_hash,
+            mortality_hash,
+        };
+        Self {
+            extra_params,
+            additional_params,
+        }
+    }
+
+    fn encode_extra_to(&self, v: &mut Vec<u8>) {
+        self.extra_params.encode_to(v);
+    }
+
+    fn encode_additional_to(&self, v: &mut Vec<u8>) {
+        self.additional_params.encode_to(v);
+    }
+}
+
+////////////////////////////////////////////////////////////
+// Third Config (using the Substrate and Polkadot Config)
+////////////////////////////////////////////////////////////
+
+pub enum StatemintConfig3 {}
```

Comment:
I don't see why you need to duplicate this config, it would be better to just to call this `PolkadotConfig` via type alias or something.

This example contains already plenty of code :)

Reply:
Just because the ExtrinsicParams are from Substrate instead of from Polkadot here. So it would be inaccurate to take just the PolkadotConfig.